### PR TITLE
feat(worker): implement alert for campaign with errors

### PR DESCRIPTION
**Pull request by stanleynguyen**
## Problem

Alert on GovSG campaign with errored messages

## Solution

Check if there're errored messages for a campaign at the finalize step in logger worker

## Deployment Checklist

_Any tasks that need to be done on the environment before releasing of this PR (e.g. Setting environment variables, running migrations, etc). If there's no task to be done, put "N/A"_

- [ ] Add `SGC_CAMPAIGN_ALERT_WEBHOOK` env var

### CODE REVIEW

**Review by jia1 — APPROVED** [review]

File: worker/src/core/loaders/message-worker/index.ts [hunk 1]
```diff
@@ -169,6 +170,45 @@ const finalize = tracer.wrap(
...
+        }
+
+        const messagesWithErr = rawMessages as Array<{
+          id: number
+          error_code: string
+          error_description: string
+        }>
+        if (
+          messagesWithErr.length === 0 ||
+          !config.get('sgcCampaignAlertChannelWebhookUrl')
```
Comment (jia1) [inline worker/src/core/loaders/message-worker/index.ts:189]: Not blocking but:

Should we be throwing when `!config.get('sgcCampaignAlertChannelWebhookUrl')`? We want to be alerted, right?

File: worker/src/core/loaders/message-worker/index.ts [hunk 2]
```diff
@@ -169,6 +170,45 @@ const finalize = tracer.wrap(
...
+          text += `| ${m.id} | ${m.error_code} | ${m.error_description} |\n`
+        })
+        text += '***********'
+        await axios
+          .post(config.get('sgcCampaignAlertChannelWebhookUrl'), {
+            text: text,
+          })
+          .catch((e) => {
+            logger.error({
+              message: 'Failed to report errors to #sgc-campaign-alerts',
```
Comment (jia1) [inline worker/src/core/loaders/message-worker/index.ts:204]: This sounds like a never-ending cycle but:

Do we get alerted through other means when this happens?

Comment (stanleynguyen) [reply worker/src/core/loaders/message-worker/index.ts:204]: we could set up an alert for this alert lol, but the log agent could go down as well 😂

Comment (jia1) [reply worker/src/core/loaders/message-worker/index.ts:204]: Ok, this won't end well...

Let's keep it as-is.

**Pull request merged**